angular/projects/spark-angular/src/lib/components/sprk-masthead/sprk-masthead.component.ts

```diff
@@ -3,397 +3,80 @@ import {
   HostListener,
   Input,
   Renderer2,
-  AfterContentInit,
+  ContentChild,
+  ElementRef,
+  AfterViewInit,
+  ViewChild,
 } from '@angular/core';
 import { Router, Event, NavigationEnd } from '@angular/router';
-import * as _ from 'lodash';
-import {
-  ISprkNarrowNavLink,
-  ISprkNarrowSelector,
-  ISprkBigNavLink,
-} from './sprk-masthead.interfaces';
+import { throttle } from 'lodash';
+import { SprkMastheadNavCollapsibleDirective } from './directives/sprk-masthead-nav-collapsible/sprk-masthead-nav-collapsible.directive';
+import { SprkMastheadBrandingDirective } from './directives/sprk-masthead-branding/sprk-masthead-branding.directive';
+import { isElementVisible } from '../../utilities/isElementVisible/isElementVisible';
```

Comment:
I took the isElementVisible method from inside this component and made it our first Angular utility. I added tests for it individually as well as making it use Renderer for the element selection instead using document.querySelector.